plugins/connectors/network_drive/plugin.go

```diff
@@ -0,0 +1,240 @@
+/* Copyright © INFINI LTD. All rights reserved.
+ * Web: https://infinilabs.com
+ * Email: hello#infini.ltd */
+
+package network_drive
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/hirochachacha/go-smb2"
+	"infini.sh/coco/modules/common"
+	"infini.sh/coco/plugins/connectors"
+	"infini.sh/framework/core/global"
+	"infini.sh/framework/core/module"
+	"infini.sh/framework/core/queue"
+	"infini.sh/framework/core/util"
+)
+
+const (
+	ConnectorNetworkDrive = "network_drive"
+	ConnectionTimeout     = time.Duration(5) * time.Second
+)
+
+// Config defines the configuration for the network drive connector.
+// It supports credentials-based scanning (for direct SMB connections).
+type Config struct {
+	//  Credentials-based Auth (SMB) Options
+	Endpoint   string   `config:"endpoint"`
+	Share      string   `config:"share"`
+	Username   string   `config:"username"`
+	Password   string   `config:"password"`
+	Domain     string   `config:"domain"` // Optional, e.g., "WORKGROUP"
+	Paths      []string `config:"paths"`
+	Extensions []string `config:"extensions"`
+}
+
+type Plugin struct {
+	connectors.BasePlugin
+	// mu protects the cancel function below.
+	mu sync.Mutex
```

Comment:
I don't quite get why we need locks here. 🫠 IMHO, we will have only 1 task for each datasource, and thus we are free from races 🤔

Reply:
that makes sense